### PR TITLE
Add dedicated spell effect to Enlarge Companion and brush up Enlarge spell effect

### LINKED ISSUE #17136: The Enlarge Companion spell should link directly to the Enlarge spell effect
As the title, when cast at 4th or higher level, the focus spell Enlarge Companion still only seems to make companions large (or does nothing if the companion is already large), but it is supposed to make them Huge. Noticible since there doesn't seem to be a way to manually change their size the way you can with monsters.

### PR DESCRIPTION
**Pull request by ammalagonc**
Closes #17136

### CODE REVIEW

File: packs/spell-effects/spell-effect-companion-enlarge.json [hunk 1]
```diff
@@ -0,0 +1,106 @@
+{
+    "_id": "r3SWKHCXxahSV54y",
+    "img": "systems/pf2e/icons/spells/enlarge-companion.webp",
+    "name": "Spell Effect: Companion Enlarge",
```
Comment (simonward) [inline packs/spell-effects/spell-effect-companion-enlarge.json:4]: Should this be "Spell Effect: Enlarge Companion"?

Comment (ammalagonc) [reply packs/spell-effects/spell-effect-companion-enlarge.json]: Oops
Sorry about that - should now be correct.

File: packs/spell-effects/spell-effect-enlarge-companion.json [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            {
+                "key": "CreatureSize",
+                "reach": {
+                    "override": "{item|flags.pf2e.rulesSelections.enlarge.reach}"
+                },
+                "resizeEquipment": true,
+                "value": "{item|flags.pf2e.rulesSelections.enlarge.size}"
+            },
+            {
+                "key": "GrantItem",
```
Comment (simonward) [inline packs/spell-effects/spell-effect-enlarge-companion.json:77]: We generally prefer in memory grants - could probably also be modified in the enlarge spell effect.

File: packs/spell-effects/spell-effect-enlarge-companion.json [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                "reach": {
+                    "override": "{item|flags.pf2e.rulesSelections.enlarge.reach}"
+                },
+                "resizeEquipment": true,
+                "value": "{item|flags.pf2e.rulesSelections.enlarge.size}"
+            },
+            {
+                "inMemoryOnly": true,
+                "key": "GrantItem",
+                "onDeleteActions": {
```
Comment (simonward) [inline packs/spell-effects/spell-effect-enlarge-companion.json:79]:
```suggestion
```

File: packs/spell-effects/spell-effect-enlarge-companion.json [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                    "grantee": "restrict"
```
Comment (simonward) [inline packs/spell-effects/spell-effect-enlarge-companion.json:80]:
```suggestion
```

File: packs/spell-effects/spell-effect-enlarge-companion.json [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                },
```
Comment (simonward) [inline packs/spell-effects/spell-effect-enlarge-companion.json:81]:
```suggestion
```

**Pull request merged**